autoPyTorch/pipeline/components/training/data_loader/base_data_loader.py

```diff
@@ -0,0 +1,271 @@
+                X=np.take(X['X_train'], X['val_indices'], axis=0),
+                y=np.take(X['y_train'], X['val_indices'], axis=0),
+                transform=val_transform
+            )
+
+        self.train_data_loader = torch.utils.data.DataLoader(
+            self.train_dataset,
+            batch_size=self.batch_size,
+            shuffle=True,
+            num_workers=X.get('num_workers', 0),
+            pin_memory=X.get('pin_memory', True),
+            drop_last=X.get('drop_last', True),
+        )
+
+        self.val_data_loader = torch.utils.data.DataLoader(
+            self.val_dataset,
+            batch_size=self.batch_size,
+            shuffle=False,
+            num_workers=X.get('num_workers', 0),
+            pin_memory=X.get('pin_memory', True),
+            drop_last=X.get('drop_last', True),
+        )
+
+        return self
+
+    def build_transform(self, X: Dict[str, Any], train: bool = True) -> torchvision.transforms.Compose:
+        """
+        Method to build a transformation that can pre-process input data
+
+        Args:
+            X (X: Dict[str, Any]): Dependencies needed by current component to perform fit
+            train (bool)" whether transformation to be built are for training of test mode"
+
+        Returns:
+            A composition of transformations
+        """
+        raise NotImplementedError()
+
+    def get_train_data_loader(self) -> torch.utils.data.DataLoader:
+        """Returns a data loader object for the train data
+
+        Returns:
+            torch.utils.data.DataLoader: A train data loader
+        """
+        assert self.train_data_loader is not None, "No train data loader fitted"
+        return self.train_data_loader
+
+    def get_val_data_loader(self) -> torch.utils.data.DataLoader:
+        """Returns a data loader object for the validation data
+
+        Returns:
+            torch.utils.data.DataLoader: A validation data loader
+        """
+        assert self.val_data_loader is not None, "No val data loader fitted"
+        return self.val_data_loader
+
+    def check_requirements(self, X: Dict[str, Any], y: Any = None) -> None:
+        """
+        A mechanism in code to ensure the correctness of the fit dictionary
+        It recursively makes sure that the children and parent level requirements
+        are honored before fit.
+
+        Args:
+            X (Dict[str, Any]): Dictionary with fitted parameters. It is a message passing
+                mechanism, in which during a transform, a components adds relevant information
+                so that further stages can be properly fitted
+        """
+
+        # make sure the parent requirements are honored
+        super().check_requirements(X, y)
+
+        # We allow reading data from a user provided dataset
+        # or from X, Y pairs
+        if 'dataset' in X and 'X_train' in X:
+            raise ValueError("Ambiguous definition of input for data loader. "
```

Comment:
I might need to change the preprocessors to also accept a dataset object, right?